lib/src/stream_extensions.dart

```diff
@@ -54,4 +54,17 @@ extension StreamExtensions<T> on Stream<T> {
     });
     return completer.future;
   }
+
+  /// Returns a view of this stream that eagerly buffers all events until
+  /// `listen()` is called.
```

Comment:
I'd drop "view of", that's not a useful description. It's a separate stream, based on this stream.

Maybe:
```dart
/// Eagerly listens to this stream and buffers events until needed.
///
/// The returned stream will emit the same events as this stream,
/// starting from when the `buffer` method is called.
/// The events are delayed until the returned stream is listened to,
/// then the buffered events will be emitted, and then further
/// events from this stream will be emitted as they arrive.
```